activity_calendar/models.py

```diff
@@ -332,15 +332,40 @@ def duration(self):
         """ Gets the duration of this activity """
         return self.end_date - self.start_date
 
-    def has_occurrence_at(self, date, is_dst_aware=False):
-        """ Whether this activity has an occurrence that starts at the specified time """
-        if not self.is_recurring:
-            return date == self.start_date
-
-        if not is_dst_aware:
-            date = util.dst_aware_to_dst_ignore(date, self.start_date, reverse=True)
-        occurences = self.get_occurrences_starting_between(date, date)
-        return list(occurences)
+    def get_occurrence_at(self, date, is_dst_aware=False):
+        """
+        Whether this activity has an occurrence that starts at the specified time
+        Note: This does not take shifts in starting moment into account!
+        :param date: Datetime instance of the occurrence
+        :param is_dst_aware: Whether the datetime instance is dst aware (relevant for recurrent activities)
+        :return:
+        """
+        # Check activitymoments on server
+        activitymoments = self.activitymoment_set.filter(recurrence_id=date)
+        if activitymoments.count() == 1:
+            return activitymoments.first()
+        elif activitymoments.count() > 1:
+            raise ActivityMoment.MultipleObjectsReturned()
```

Comment:
Is this even possible? Each recurrence_id, parent_activity pair is unique.

Reply:
Nope, it shouldn't. Hadn't realised that.